tests/FunctionalTestCase.php

```diff
@@ -147,6 +150,78 @@ protected function assertCollectionCount($namespace, $count): void
         $this->assertEquals($count, $document['n']);
     }
 
+    /**
+     * Asserts that a collection with the given name does not exist on the
+     * server.
+     *
+     * If $databaseName is not specified, the current database will be used.
+     *
+     * @param string $collectionName
+     * @param string $databaseName
+     */
+    protected function assertCollectionDoesNotExist(string $collectionName, ?string $databaseName = null): void
+    {
+        if (! isset($databaseName)) {
+            $databaseName = $this->getDatabaseName();
+        }
+
+        $operation = new ListCollections($this->getDatabaseName());
+        $collections = $operation->execute($this->getPrimaryServer());
+
+        $foundCollection = null;
+
+        foreach ($collections as $collection) {
+            if ($collection->getName() === $collectionName) {
+                $foundCollection = $collection;
+                break;
+            }
+        }
+
+        $this->assertNull($foundCollection, sprintf('Collection %s exists', $collectionName));
+    }
+
+    /**
+     * Asserts that a collection with the given name exists on the server.
+     *
+     * If $databaseName is not specified, the current database will be used.
+     * An optional $callback may be provided, which should take a CollectionInfo
+     * argument as its first and only parameter. If a CollectionInfo matching
+     * the given name is found, it will be passed to the callback, which may
+     * perform additional assertions.
+     *
+     * @param string   $collectionName
+     * @param string   $databaseName
```

Comment:
Pending @alcaeus' input, this can be changed to either `?string` or `string|null`.

Reply:
@tanlisu: Let's just delete both of these `@param` lines.